compiler/rustc_codegen_llvm/src/va_arg.rs

```diff
@@ -946,23 +947,78 @@ pub(super) fn emit_va_arg<'ll, 'tcx>(
             AllowHigherAlign::Yes,
             ForceRightAdjust::No,
         ),
+        Arch::LoongArch32 => emit_ptr_va_arg(
+            bx,
+            addr,
+            target_ty,
+            if target_ty_size > 2 * 4 { PassMode::Indirect } else { PassMode::Direct },
+            SlotSize::Bytes4,
+            AllowHigherAlign::Yes,
+            ForceRightAdjust::No,
+        ),
+        Arch::LoongArch64 => emit_ptr_va_arg(
+            bx,
+            addr,
+            target_ty,
+            if target_ty_size > 2 * 8 { PassMode::Indirect } else { PassMode::Direct },
+            SlotSize::Bytes8,
+            AllowHigherAlign::Yes,
+            ForceRightAdjust::No,
+        ),
```

Comment:
https://github.com/llvm/llvm-project/blob/c386d6d3bfa400e94ac9cadb85a0b1a72ed54b2b/clang/lib/CodeGen/Targets/LoongArch.cpp#L434

zero-sized types can't reach this logic currently, but we could handle them. This logic is very similar for riscv, but it has an annoying exception for the 32-bit embedded ABI which lowers alignment for `f64` and `i64/u64` from 8 to 4. Apparently that may change as things are ratified.

Reply:
Hrrrm. Any chance we can error if we encounter a case we expect to be changed later, and let the loongarch maints fix it when they have more confidence?

Or we can just poke them now, I guess. Hey @heiher!

Reply:
LGTM. I've confirmed that on the LoongArch32 ABI, `f64` and `i64`/`u64` are still 8-byte aligned. Thanks!

Reply:
Thanks for checking that!